v2/types.go

```diff
@@ -133,13 +133,13 @@ func (walletAccount *WalletAccount) SignTransactionHash(hashToSign *TransactionH
 	if walletAccount.Address == nil || walletAccount.Keys == nil {
 		return nil, errors.New("'AccountAddress' or 'Keys' field is not initialized or empty")
 	}
-	if walletAccount.Keys.Keys == nil || len(walletAccount.Keys.Keys) == 0 {
+	if len(walletAccount.Keys.Keys) == 0 {
 		return nil, errors.New("'WalletAccount.Keys' is not initialized or empty")
 	}
 
 	signaturesMap := make(map[uint8]*AccountSignatureMap, int(walletAccount.Keys.Threshold.Value))
 	for credIdx, credData := range walletAccount.Keys.Keys {
-		if credData.Keys == nil || len(credData.Keys) == 0 {
+		if len(credData.Keys) == 0 {
```

Comment:
so I assume `len(nil) == 0`?

Reply:
Yes exactly. `staticcheck` complained that there is no need to check for `nil` when working with maps/lists and checking their length, since `len(nil) == 0`, so you can just get the length directly without checking for nil first.